### DISCUSSION TITLE
The use of seeding / patterns for improving a NER model

### DISCUSSION THREAD

**Discussion by Larsdegroot** [explosion/spaCy#10900] [Help: Coding & Implementations]
Hi all,

im currently tasked with improving a NER model with 'seeding' it with a list of patterns. per example:

`{"label": "PHENOTYPE", "pattern": [{"LOWER": "mitochondrion"}]}`

the assumption was that adding patterns like this during training would give the model an 'idea' / examples of what the entity phenotype looks like. However i'm finding no way to do this. 

What i am finding is using patterns to match entities in corpus you're going to annotate and then manually accepting / rejecting the entities that were found by the simple pattern match. 
in this context seeding / providing a pattern file doesn't do anything directly with the model it just improves the annotating workflow. 

Am i correct in that it is not possible to improve a model by seeding alone during training and that it's just a tool for easier data annotation?

**Comment by mapadofu** [comment]
I have what I believe a very closely related question here: https://github.com/explosion/spaCy/discussions/10894

Though that is a bit more focused on the mechanics of incorporating labels generated by a rules-based component into the NER model training process.

**Comment by polm** [comment]
You can use a rule-based annotator to create training data which can use for a model, that process is called "weak supervision". You might look at [skweak](https://github.com/NorskRegnesentral/skweak), which is for weak supervision in spaCy.

There's not an easy to way to tag arbitrary spans as being "probably entities", the way you might use existence in a glossary as a feature in some older NER systems.
